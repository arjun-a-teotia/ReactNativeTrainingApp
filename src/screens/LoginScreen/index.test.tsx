/**
 * @format
 */

import 'react-native';
import React from 'react';
import {LoginScreen} from './index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import analytics from '@react-native-firebase/analytics';
import {Alert} from 'react-native';
const mockNavigate = jest.fn();

jest.spyOn(Alert, 'alert');
jest.mock('@react-navigation/native', () => {
  const actual = jest.requireActual('@react-navigation/native');
  return {
    ...actual,
    useNavigation: () => ({
      navigate: mockNavigate,
      dispatch: mockNavigate,
    }),
  };
});

jest.mock('@react-native-firebase/app', () => {
  return () => ({
    app: jest.fn(),
  });
});
jest.mock('@react-native-firebase/auth', () => {
  return jest.fn().mockReturnValue({
    signInWithEmailAndPassword: jest.fn(() => Promise.resolve({data: {}})),
  });
});
jest.mock('@react-native-firebase/analytics', () => {
  return jest.fn().mockReturnValue({logEvent: jest.fn()});
});
jest.mock('react-native-qrcode-svg', () => ({}));
jest.useFakeTimers();

describe('Login Screen Testing', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('navigates to register screen clicking Register button', async () => {
    const {getByTestId} = render(<LoginScreen />);
    fireEvent.press(getByTestId('Register'));
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
  it('Login button click', async () => {
    const {getByTestId} = render(<LoginScreen />);
    // const mock = jest.fn();
    // fireEvent.changeText(getByTestId('email'), 'test');
    // expect(mock).toHaveBeenCalledWith('test');
    // fireEvent.changeText(getByTestId('password'), 'test');
    // expect(mock).toHaveBeenCalledWith('test');
    fireEvent.press(getByTestId('Login'));
    await waitFor(async () => {
      expect(analytics().logEvent).toBeCalledTimes(1);
    });
  });
});
