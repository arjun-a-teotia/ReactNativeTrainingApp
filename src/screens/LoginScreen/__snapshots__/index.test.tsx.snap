// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Testing renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "paddingHorizontal": 30,
      "paddingTop": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Email"
      style={
        Object {
          "fontSize": 32,
          "marginTop": 20,
          "width": "80%",
        }
      }
      testID="email"
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Password"
      secureTextEntry={true}
      style={
        Object {
          "fontSize": 32,
          "marginTop": 20,
          "width": "80%",
        }
      }
      testID="password"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Login"
    >
      <Text
        style={
          Object {
            "color": "steelblue",
            "fontSize": 32,
            "marginTop": 50,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "opacity": 1,
        "textAlign": "right",
      }
    }
    testID="Register"
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginTop": 50,
        }
      }
    >
      Register
    </Text>
  </View>
</RNCSafeAreaView>
`;
